src/Circlereader/Circlereader.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as flatbuffers from 'flatbuffers';
+import {readFileSync} from 'fs';
+
+import {BuiltinOptions} from './circle-analysis/circle/builtin-options';
+import {Model} from './circle-analysis/circle/model';
+import {ModelOperator, Node, NodeAttributes, NodeInput, NodeOutput, NodeProperties} from './type/types';
+import {getNodeInputs} from './util/getNodeInputs';
+import {getNodeOutputs} from './util/getNodeOutputs';
+import {initBuiltInOperator} from './util/initBuiltinOptions';
+import {initSubgraphNodes} from './util/initSubgraphNodes';
+import {setAttributesByOption} from './util/setAttributesByOption';
+
+export function decoder(path: string) {
+  const file = readFileSync(path);
+  const buffer = file instanceof Uint8Array ? file : new Uint8Array(file);
+  const bb = new flatbuffers.ByteBuffer(buffer);
+  const model = Model.getRootAsModel(bb);
+  const subgraphsLength = model.subgraphsLength();
+
+  let resultArr: Array<ModelOperator> = [];
+  let builtInOperatorArr: Array<String> = initBuiltInOperator(model);
+
+  // init subgraphs
+  let subgraph = model.subgraphs(0);
+
+  if (subgraph !== null) {
```

Comment:
I think this condition is redundant because `for (let subgraphIdx = 0; subgraphIdx < subgraphsLength; subgraphIdx++) {` handles it :)

Reply:
Ok. I'll remove it :) Thank u for ur review.

Reply:
If you delete line 42,
I am getting an error in `model.subgraphs(subgraphIdx, subgraph)` of line 44. 
`Subgraph | null` type cannot be assigned to `Subgraph | undefined` type. So, it seems impossible to fix it.

Reply:
Hmm... I am surprised because such case is occurred.
Usually, normal circle model has at least one subgraph.
When such case occurred...?

Reply:
If you do not handle exceptions in typescript, the above problem occurs. If the data is guaranteed to exist, you can avoid errors through `optional chaining`.